src/traccuracy/metrics/_track_overlap.py

```diff
@@ -0,0 +1,108 @@
+"""This submodule implements routines for Track Purity (TP) and Target Effectiveness (TE) scores.
+
+Definitions (Bise et al., 2011; Chen, 2021; Fukai et al., 2022):
+
+- TE for a single ground truth track T^g_j is calculated by finding the predicted track T^p_k
+  that overlaps with T^g_j in the largest number of the frames and then dividing
+  the overlap frame counts by the total frame counts for T^g_j.
+  The TE for the total dataset is calculated as the mean of TEs for all ground truth tracks,
+  weighted by the length of the tracks.
+
+- TP is defined analogously, with T^g_j and T^p_j being swapped in the definition.
+"""
+
+from itertools import product
+from typing import TYPE_CHECKING, Any, List, Tuple
+
+from traccuracy._tracking_graph import TrackingGraph
+
+from ._base import Metric
+
+if TYPE_CHECKING:
+    from ._base import Matched
+
+
+class TrackOverlapMetrics(Metric):
+    supports_many_to_one = True
+
+    def __init__(self, matched_data: "Matched", include_division_edges: bool = True):
+        """Calculate metrics for longest track overlaps.
+
+        - Target Effectiveness
+        - Track Purity
+
+        Args:
+            matched_data (Matched): Matched object for set of GT and Pred data
+            include_division_edges (bool, optional): If True, include edges at division.
+        """
```

Comment:
In order for the docs to render correctly, can you move this docstring under the class (line 26) itself instead of under init?

Reply:
I moved the __init__ docstring to under the class!